pkg/web_app/lib/src/account.dart

```diff
@@ -23,13 +24,69 @@ late final useNewSignin = _newSigningMetaContent == '1';
 void setupAccount() {
   if (useNewSignin) {
     // TODO: review client-side sign-in methods.
+    _initSessionMonitor();
     _initWidgets();
     return;
   } else {
     _setupOldAccount();
   }
 }
 
+void _initSessionMonitor() {
+  // No need to monitor session on non-editable pages,
+  // where no action or state-change may be started.
+  if (!pageData.isSessionAware) {
+    return;
+  }
+
+  final checkFrequency = Duration(minutes: 5);
```

Comment:
It might be better to just make a `while` loop and then do `await Future.delayed(sessionExpiresThreshold.subtract(now.subtract(authenticatedAt)))`

or something like that. Since there is no need to refresh in the meantime.

Reply:
Added a TODO for it.